### DISCUSSION TITLE
Deploying 2 Strimzi operators with different versions

### DISCUSSION THREAD

**Discussion by mabruch123** [strimzi/strimzi-kafka-operator#9077] [General]
Hi,
I have a problem that when I upgrade the cluster operator it does rolling update for all of the Kafkas and I dont want that.
I want to deploy 2 different operators in the same openshift cluster;
One will manage Kafkas which I dont care if they do a rolling update when upgrading the cluster operator and one will mange those who I don't want auto rolling update.
In this case, I will first upgrade only one of the operators with the new CRD's (which are clusterwide).
In this scenario one of the operators will use ADVANCED CRD's of the next version operator version (because I haven't upgraded it).
My question is if this could cause problems or should I assume that generally an old operator can work with new CRD's of advanced operator version?

**Comment by scholzj** [comment]
OpenShift (Kubernetes) is not multi-tenant when it comes to operators. The best way to solve this is to have separate OpenShift clusters for the Kafka's you _care_ about and those you _don't care_ about. 

If you want to run two operators in parallel, you can do it in most cases by using the _newer_ CRDs. But this is not something that can be guaranteed for any two versions - the problems might happen for example next time we will be changing the API version. So you should make sure to test it in some test environment first. You should also make sure that only one operator manages given Kafka cluster.

Also, keep in mind the risks of running multiple operators in parallel. If the shared resources are deleted by mistake, they can easily take down all your cluster etc.
